File: Array/Python/subArraysumDivisible.py
from collections import defaultdict
nums = [4,5,0,-2,-3,1]

k = 5

sum,count, mp = 0,0, defaultdict(int, {0:1})

for i in nums:
    sum += i
    rem = sum%k
    count += mp[rem]
    mp[rem] += 1

print(count)

File: Array/C++/subArraysumDivisible.cpp
#include <bits/stdc++.h>
#define log(x) cout<<x<<endl;
using namespace std;

int main()
{
    vector<int> nums = {4,5,0,-2,-3,1}; int k=5;
    unordered_map<int, int> mp;
    int sum=0, count=0;
    mp[0]++;

    for(auto i: nums){
        sum += i;
        if(mp[(sum%k +k)%k]>0){
            count+= mp[((sum%k+k)%k)];
        }
        mp[((sum%k+k)%k)]++;
    }
    for(auto i: mp){
        cout<<i.first<<" "<<i.second<<endl;
    }
    log(count)
}
